datafusion/substrait/src/logical_plan/consumer.rs

```diff
@@ -434,6 +434,16 @@ pub async fn from_substrait_rel(
                 None => plan_err!("JoinRel without join condition is not allowed"),
             }
         }
+        Some(RelType::Cross(cross)) => {
+            let left: LogicalPlanBuilder = LogicalPlanBuilder::from(
+                from_substrait_rel(ctx, cross.left.as_ref().unwrap(), extensions).await?,
+            );
+            let right = LogicalPlanBuilder::from(
+                from_substrait_rel(ctx, cross.right.as_ref().unwrap(), extensions)
+                    .await?,
+            );
+            left.cross_join(right.build()?)?.build()
```

Comment:
`LogicalPlanBuilder` to right side hand plan is unnecessary -- You create a builder from a plan and freeze it to plan immediately.

Reply:
get it! thanks @waynexia